src/libstd/primitive_docs.rs

```diff
@@ -140,9 +142,59 @@ mod prim_bool { }
 /// [`Ok`] variant. This illustrates another behaviour of `!` - it can be used to "delete" certain
 /// enum variants from generic types like `Result`.
 ///
+/// ## Infinite loops
+///
+/// While [`Result<T, !>`] is very useful for removing errors, `!` can also be used to removed
+/// successes as well. If we think of [`Result<T, !>`] as "if this function returns, it has not
+/// errored," we get a very intuitive idea of [`Result<!, E>`] as well: if the function returns, it
+/// *has* errored.
+///
+/// For example, consider the case of a simple web server, which can be simplified to:
+///
+/// ```ignore (hypothetical-example)
```

Comment:
I've never seen this notation, does this work?

Reply:
I actually copied it from an example earlier in the file, so, I assume so?

Reply:
It doesn't do anything. It's recognized as non-rust code. That's pretty much all.